packages/runtime/runtime-utils/src/objectstorageutils.ts

```diff
@@ -19,24 +19,23 @@ export function getNormalizedObjectStoragePathParts(path: string) {
     return [];
 }
 
-export async function listBlobsAtTreePath(inputTree: ITree, path: string): Promise<string[]> {
+export async function listBlobsAtTreePath(inputTree: ITree | undefined, path: string): Promise<string[]> {
     const pathParts = getNormalizedObjectStoragePathParts(path);
     let tree: ITree | undefined = inputTree;
     while (tree?.entries !== undefined && pathParts.length > 0) {
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        const part = pathParts.shift()!;
-        const index = tree.entries.findIndex((value) => {
+        const part = pathParts.shift();
+        const treeEntry = tree.entries.find((value) => {
             if (value.type === "Tree" && value.path === part) {
                 return true;
             } else {
                 return false;
             }
         });
-        if (index === -1) {
-            tree = undefined;
-        } else {
-            const treeEntry = tree.entries[index];
+
+        if (treeEntry?.type === "Tree") {
```

Comment:
if treeEntry isn't undefined, treeEntry's type should always be "Tree" already, right?  why double checking here?

Reply:
The check is unfortunately redundant, but the type of `treeEntry` is actually a union, so it could be either a tree, a blob, or an attachment. The above check in the arrow function does already imply that it is a tree, but the type system does not know this because there is no type assertion. The function could be modified to include a type assertion, but the type it is asserting would be unpleasant to redeclare solely for the purpose of such an assertion.

Reply:
let's put a comment explaining the redundant check so the next person isn't confused.